File: src/main/java/com/und/security/AuthenticationUtils.java
package com.und.security;

import org.springframework.security.access.AccessDeniedException;
import org.springframework.security.authentication.AnonymousAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContext;
import org.springframework.security.core.context.SecurityContextHolder;

public class AuthenticationUtils {

  public static boolean isUserLoggedIn() {
    SecurityContext securityContext = SecurityContextHolder.getContext();
    if (securityContext == null) {
      return false;
    }

    Authentication authentication = securityContext.getAuthentication();
    if (authentication == null || !authentication.isAuthenticated() || (authentication instanceof AnonymousAuthenticationToken)) {
      return false;
    }
    return true;
  }

  public static Object getPrincipal() {
    SecurityContext securityContext = SecurityContextHolder.getContext();
    if (securityContext == null) {
      throw new AccessDeniedException("User is not logged in to the system.");
    }

    Authentication authentication = securityContext.getAuthentication();
    if (authentication == null) {
      throw new AccessDeniedException("User is not logged in to the system.");
    }

    Object principal = authentication.getPrincipal();
    return principal;
  }


  public static String getName() {
    Object principal = getPrincipal();
    if (principal instanceof EventUser) {
      return ((EventUser) principal).getUsername();
    }
    return null;
  }

  public static String getId() {
    Object principal = getPrincipal();
    if (principal instanceof EventUser) {
      return "" + ((EventUser) principal).getId();
    }
    return "_";
  }


}

File: src/main/kotlin/com/und/security/AuthenticationUtils.kt
package com.und.security

import org.springframework.security.access.AccessDeniedException
import org.springframework.security.authentication.AnonymousAuthenticationToken
import org.springframework.security.core.Authentication
import org.springframework.security.core.context.SecurityContext
import org.springframework.security.core.context.SecurityContextHolder

object AuthenticationUtils {

    val isUserLoggedIn: Boolean
        get() {
            val securityContext = SecurityContextHolder.getContext() ?: return false

            val authentication = securityContext.authentication
            return if (authentication == null || !authentication.isAuthenticated || authentication is AnonymousAuthenticationToken) {
                false
            } else true
        }

    val principal: Any
        get() {
            val securityContext = SecurityContextHolder.getContext() ?: throw AccessDeniedException("User is not logged in to the system.")

            val authentication = securityContext.authentication ?: throw AccessDeniedException("User is not logged in to the system.")

            return authentication.principal
        }


    val name: String?
        get() {
            val principal = principal
            return (principal as? UndUserDetails)?.username
        }

    val id: String
        get() {
            val principal = principal
            return if (principal is UndUserDetails) {
                "" + principal.id!!
            } else "_"
        }


}